scripts/test-addon-local.sh

```
@@ -0,0 +1,297 @@
+else
+    fail "repository.yaml missing (HA can't discover the addon repo)"
+fi
+
+# ---- Optional: Docker build ----
+if [ "${1:-}" = "--build" ]; then
+    echo ""
+    echo "=== Docker Build Test ==="
+
+    # Simulate CI: copy source into build context
+    TMPCTX=$(mktemp -d)
+    trap "rm -rf $TMPCTX" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)